Lib/test/test_importlib/metadata/fixtures.py

```diff
@@ -44,6 +44,16 @@ def save_cwd():
         os.chdir(orig)
 
 
+@contextlib.contextmanager
+def save_mode(path, *, follow_symlinks=True):
+    path = pathlib.Path(path)
+    orig = path.stat(follow_symlinks=follow_symlinks)
+    try:
+        yield
+    finally:
+        path.chmod(orig.st_mode, follow_symlinks=follow_symlinks)
+
+
```

Comment:
Tests in this path (`test_importlib/metadata`) are for the `importlib.metadata` package. Please move this to a new `test_importlib.fixtures` or another suitable location outside `metadata`. Thanks.

Reply:
Is it okay for `test.test_importlib.import_.test_path` to use _existing_ support functions from here? (e.g. `test.test_importlib.metadata.fixtures.save_cwd()`)

Reply:
Oh. Good point. I suspect your usage is probably the first such instance (though I haven't checked).

My motivation for keeping them separate is mainly to support the backports (i.e. [`importlib_metadata`](/python/importlib_metadata)), which share the same code. Since the relationship between this code and the backports are fairly stable, I'm okay with other stdlib tests building on these fixtures. However, there may be suitable fixtures in `test.support**` (e.g. `test.support.os_helper.change_cwd` and `test.support.os_helper.temp_dir`) and those should be preferred. Can you explore using those instead?

In the meantime, I've updated https://github.com/python/cpython/issues/120801 to track consolidating the fixtures. If you do end up relying on the metadata fixtures, I'll probably end up replacing those in that effort, so if it's not too much a hassle, please consider using those. Thanks.

Reply:
In #120803, I'm proposing to remove those fixtures. Please let me know if you'd like me to help adapt this PR to accommodate those changes.

Reply:
I'd wondered if there was appetite for consolidating such fixtures and decorators across the test suite. So will do.

Reply:
Thanks!